openedx/features/survey_report/api.py

```diff
@@ -13,23 +13,34 @@
     get_unique_courses_offered
 )
 
-
-def generate_report() -> None:
-    """ Generate a report with relevant data."""
+def get_report_data() -> dict:
+    """ Get data from database to generate a new report."""
     courses_offered = get_unique_courses_offered()
     learners = get_recently_active_users(weeks=3)
     registered = get_registered_learners()
     certificates = get_generated_certificates()
     enrollments = get_course_enrollments()
     extra_data = settings.SURVEY_REPORT_EXTRA_DATA
 
-    survey_report = SurveyReport(
-        courses_offered=courses_offered,
-        learners=learners,
-        registered_learners=registered,
-        generated_certificates=certificates,
-        enrollments=enrollments,
-        extra_data=extra_data,
-    )
+    return {
+        "courses_offered": courses_offered,
+        "learners": learners,
+        "registered_learners": registered,
+        "generated_certificates": certificates,
+        "enrollments": enrollments,
+        "extra_data": extra_data,
+    }
+
+def generate_report(defaults:bool=False) -> int:
+    """ Generate a report with relevant data."""
+    data = {}
+    if not defaults:
+        data = get_report_data()
 
+    survey_report = SurveyReport(**data)
     survey_report.save()
+    return survey_report.id
+
+
+def update_report(survey_report_id: int, data:dict) -> None:
+    SurveyReport.objects.filter(id=survey_report_id).update(**data)
```

Comment:
why are you changing this here? is it needed? I thought this only added the django admin view

Reply:
@mariajgrimaldi this is part of the first part of the project "You should have the option to generate a new report manually", currently, you just can do it by CLI.

Reply:
Perhaps, we can just add the admin site just with view option and add the generate button in a future PR, but I don't know.